src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -663,48 +663,79 @@ protected virtual Task<ReturnType> InvokeCoreAsync<ReturnType>(int? id, string t
         }
 
         /// <summary>
-        /// Creates a dictionary which maps a request method name to its clr method name via JsonRpcMethodAttribute value.
+        /// Creates a dictionary which maps a request method name to its clr method name via <see cref="JsonRpcMethodAttribute" /> value.
         /// </summary>
-        /// <param name="target"></param>
-        /// <returns></returns>
+        /// <param name="target">Object to reflect over and analyze its methods.</param>
+        /// <returns>Dictionary which maps a request method name to its clr method name.</returns>
         private static Dictionary<string, string> GetRequestMethodToClrMethodMap(object target)
         {
-            // First we create a dictionary to map clr method names to JsonRpcmethodAttribute value, this simplifies the process of detecting attribute name conflicts.
-            var methodToAttributeMap = new Dictionary<string, string>(StringComparer.Ordinal);
+            // Create a dictionary to map clr method names to <see cref="JsonRpcMethodAttribute" /> values, used to detect that same method shouldn't have multiple attribute values.
+            var clrMethodToRequestMethodMap = new Dictionary<string, string>(StringComparer.Ordinal);
+
+            // Create a dictionary to map <see cref="JsonRpcMethodAttribute" /> values to clr method names, used to detect that the same attribute value doesn't map to multiple methods.
+            var requestMethodToClrMethodMap = new Dictionary<string, string>(StringComparer.Ordinal);
+
             if (target != null)
             {
                 for (TypeInfo t = target.GetType().GetTypeInfo(); t != null; t = t.BaseType?.GetTypeInfo())
                 {
+                    HashSet<string> clrMethodsWithoutReplacementName = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
What if instead of maintaining this 3rd collection, you simply added all methods to your two dictionaries regardless whether they have attributes applied or not? Then you'll detect the conflicts you mean to, I *think*.
Not a required change. Just an idea.